CMakeLists.txt

```diff
@@ -821,3 +826,32 @@ IF(CPUINFO_SUPPORTED_PLATFORM AND CPUINFO_BUILD_TOOLS)
     INSTALL(TARGETS cpuid-dump RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
   ENDIF()
 ENDIF()
+
+# ---[ pkg-config manifest. This is mostly from JsonCpp...
+if(CPUINFO_BUILD_PKG_CONFIG AND (CPUINFO_LIBRARY_TYPE STREQUAL "default" OR CPUINFO_LIBRARY_TYPE STREQUAL "shared"))
```

Comment:
Does pkg-config work with only shared libraries? If so, the right approach would be to report a warning if non-shared library is being built with `CPUINFO_BUILD_PKG_CONFIG` is `TRUE` and set `CPUINFO_BUILD_PKG_CONFIG` to `FALSE`.

Reply:
It can work with [static libraries](https://people.freedesktop.org/~dbn/pkg-config-guide.html) as well via `--static`.

Reply:
Then why is this check needed at all?

Reply:
Good point. Generally static libraries are bad engineering practice these days in UNIX like environments. I figured nobody wanted it if they are building a static library, but I'll modify the check.